module/zfs/vdev_label.c

```diff
@@ -1184,7 +1184,18 @@ vdev_uberblock_compare(const uberblock_t *ub1, const uberblock_t *ub2)
 	if (likely(cmp))
 		return (cmp);
 
-	return (AVL_CMP(ub1->ub_timestamp, ub2->ub_timestamp));
+	cmp = AVL_CMP(ub1->ub_timestamp, ub2->ub_timestamp);
+	if (likely(cmp))
+		return (cmp);
+
+	/*
+	 * XXX What if !MMP_VALID(initial_ub) and MMP_VALID(later_ub)?
+	 */
```

Comment:
I don't think we should try and handle this case.  Both uberblocks need to have valid sequence number, otherwise they're considered equivalent as is the case today.  Something like this, with a reasonable comment.

```c
if (MMP_VALID(ub1) &&  MMP_SEQ_VALID(ub1) &&
    MMP_VALID(ub2) && MMP_SEQ_VALID(ub2)) {
        return (AVL_CMP(MMP_SEQ(ub1), MMP_SEQ(ub2)));
} else {
        return (0);
}
```

Reply:
I think that we should treat no MMP and/or no SEQ as 0.  If one uberblock has valid MMP and SEQ, and the other does not, but both have the same txg and timestamp, they were written by different hosts.  I think we would see this if host A (new ZFS) starts an activity test, and during the test, host B (new ZFS) imports the pool and writes out an uberblock, and the pool was last written by an older version of ZFS.

Reply:
Although this is admittedly a corner case.

Reply:
Actually, I think what I wrote makes no sense.  It looks from spa_import->spa_config_update() like we immediately sync out the mosconfig after an import, so txg changes almost right away.  Am I reading that correctly?  In that case an import occurring on a remote host while we are in activity_check locally would show up because txg would change.

Reply:
That's true, and the updated code and comment make sense.  I'd only ask that you move the `seq1` and `seq2` declarations under this comment for clarity, we do allow inline declarations.

Reply:
Done.